packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-embed.php

```diff
@@ -0,0 +1,289 @@
+	 * @return string Label for the provider.
+	 */
+	private function get_provider_label( string $provider, array $attr ): string {
+		// Use custom label if provided.
+		if ( ! empty( $attr['label'] ) ) {
+			return $attr['label'];
+		}
+
+		// Use default label based on provider.
+		switch ( $provider ) {
+			case 'spotify':
+				return __( 'Listen on Spotify', 'woocommerce' );
+			case 'soundcloud':
+				return __( 'Listen on SoundCloud', 'woocommerce' );
+			case 'pocket-casts':
+				return __( 'Listen on Pocket Casts', 'woocommerce' );
+			case 'mixcloud':
+				return __( 'Listen on Mixcloud', 'woocommerce' );
+			case 'reverbnation':
+				return __( 'Listen on ReverbNation', 'woocommerce' );
+			default:
+				return __( 'Listen to the audio', 'woocommerce' );
+		}
+	}
+
+	/**
+	 * Render a simple link fallback for non-audio embeds.
+	 *
+	 * @param array             $attr Block attributes.
+	 * @param string            $block_content Block content.
+	 * @param array             $parsed_block Parsed block.
+	 * @param Rendering_Context $rendering_context Rendering context.
+	 * @return string Rendered link or empty string if no valid URL.
+	 */
+	private function render_link_fallback( array $attr, string $block_content, array $parsed_block, Rendering_Context $rendering_context ): string {
+		// Try to get URL from attributes first.
+		$url = $attr['url'] ?? '';
+
+		// If no URL in attributes, try to extract from block content.
+		if ( empty( $url ) ) {
+			// Look for any HTTP/HTTPS URL in the content with proper boundaries.
+			if ( preg_match( '/(?<![a-zA-Z0-9.-])https?:\/\/[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}[a-zA-Z0-9\/?=&%-]*(?![a-zA-Z0-9.-])/', $block_content, $matches ) ) {
+				$url = $matches[0];
+			}
+		}
+
+		// If still no URL, try to use provider-specific base URL if we have a provider.
+		if ( empty( $url ) && isset( $attr['providerNameSlug'] ) ) {
+			$url = $this->get_provider_base_url( $attr['providerNameSlug'] );
+		}
+
+		// Validate URL with both filter_var and wp_http_validate_url.
+		if ( empty( $url ) || ! filter_var( $url, FILTER_VALIDATE_URL ) || ! wp_http_validate_url( $url ) ) {
+			return '';
+		}
+
+		// Get link text - use custom label if provided, otherwise use provider label for base URLs or URL.
+		if ( ! empty( $attr['label'] ) ) {
+			$link_text = $attr['label'];
+		} else {
+			// Check if this is a provider base URL (like https://open.spotify.com/).
+			$provider = $attr['providerNameSlug'] ?? '';
+			$base_url = $this->get_provider_base_url( $provider );
+
+			if ( ! empty( $base_url ) && $url === $base_url ) {
+				// Use provider-specific label for base URLs.
+				$link_text = $this->get_provider_label( $provider, $attr );
+			} else {
+				// Use the URL itself for specific URLs.
+				$link_text = $url;
+			}
+		}
+
+		// Create a simple link.
+		$link_html = sprintf(
+			'<a href="%s" target="_blank" rel="noopener nofollow" style="color: #0073aa; text-decoration: underline;">%s</a>',
```

Comment:
Would it make sense to read the color from the email attributes instead of using a hardcoded color?